### DISCUSSION TITLE
Mail delivery not working with on premise installation

### DISCUSSION THREAD

**Discussion by destracke** [outline/outline#2172] [Self Hosting]
Hello everyone,
first of alle thank you very much for this awesome software!

I deployed an on premise installation via docker-compose, using environment production-ssl-disabled since I use an apache2 reverse proxy on the docker host.

No matter what I configure in the SMTP environment settings of Outline, it doesn't work:
1) Delivery via postfix of docker host results in _(node:60) UnhandledPromiseRejectionWarning: Error: connect ECONNREFUSED 127.19.0.1:587_

2) Delivery via outside smtp server (gmail, ethereal, etc.) results in: _(node:60) UnhandledPromiseRejectionWarning: Error: 139753565830472:error:1408F10B:SSL routines:ssl3_get_record:wrong version number:../deps/openssl/openssl/ssl/record/ssl3_record.c:332:_

Does anybody have any clues how to get it working?

Best regards
Dennis

**Comment by tommoor** [comment]
It seems like these hosts require the TLS cipher to be set explicitly, see issue: https://github.com/nodemailer/nodemailer/issues/165

This will require a patch to fix as this value is not possible to set in the current image.

**Reply by tommoor** [reply]
Created an issue to include a setting for this in the next release: https://github.com/outline/outline/issues/2175

**Reply by destracke** [reply]
Thanks for the answer. I would favor solution 1 of my first post. Since writing my initial post, I could solve my ECONNREFUSED error.

Here's my mail config for outline:
_SMTP_HOST=docker-host
SMTP_PORT=25
SMTP_USERNAME=
SMTP_PASSWORD=
SMTP_FROM_EMAIL=outline@****
SMTP_REPLY_EMAIL=outline@****_

Outline can now successfully reach the postfix installation on my host, but throws the same error 
_(node:60) UnhandledPromiseRejectionWarning: Error: 139753565830472:error:1408F10B:SSL routines:ssl3_get_record:wrong version number:../deps/openssl/openssl/ssl/record/ssl3_record.c:332:_

My postfix allows unencrypted smtp delivery. Is there a way to enable this for outline? When I look into the source code in
_/server/mailer.js_, I can see that _secure_ should be false for me, because my _NODE_ENV_ is equal to _production-ssl-disable_.

Maybe the option _ignoreTLS_ from nodemailer could help? (https://nodemailer.com/smtp/)

Best regards
Dennis
